### DISCUSSION TITLE
How to use history API to get all emails sent

### DISCUSSION THREAD

**Discussion by xmw2003** [emailjs-com/help#98] [Q&A]
I am trying to get the entire history of emails sent using their API, and I'm a little lost on how to do this? I tried to do a loop until the response json is_last_page == true, but it seems to loop forever and I know that there are not that many emails in the history. How should I be doing this? I made sure to also wait the 1 second per request.

**Comment by xr0master** [comment]
The logic is correct, but we don't know how you executed it.

**Comment by xmw2003** [comment]
I just replaced the public and private key placeholders with my own, and
ran it using node.js in VScode terminal. I also tried without the page at
the end and then the loop runs forever. It does return one page, but I know
there are far more emails than one page of results.

**Reply by xr0master** [reply]
I apologize for the late response. Please note that the account subscription has a data retention time. Data is NOT stored beyond this time. The Email History was created to effectively solve issues with the emails deliverability and debugging, but not to store clients' private sensitive data.

No data, no leaks.
